application/Synchronization/SyncManager.cs

```diff
@@ -0,0 +1,269 @@
+            {
+                _interval = _intervalDefault;//set default delay between updates
+
+                _logger.WriteEntry($"Update Interval was set to its default value ({_intervalDefault} hours)", LoggingLevel.Info);
+            }
+                
+            //check amazon keys
+            //check dbcontext
+        }
+        
+        private void ExecuteUpdate(Object state)
+        {
+            if (!_isIdle) return;//if still updating products do nothing
+
+            _isIdle = false;//set the sync manager state to active
+
+            //get all the products at once ==> if product count is high it could take time
+            //get products by last updated
+            using (var dbContext = new BagsContext(_config))
+            {
+                _logger.WriteEntry("##########################################################", LoggingLevel.Info);
+                _logger.WriteEntry($"########## Update #{_updatesCount} Started ##########", LoggingLevel.Info);
+
+                var summary = new UpdateSummary()
+                {
+                    //ProductCount = products.Count,
+                    StartDate = DateTime.Now
+                };
+
+                var startIndex = 0;
+
+                //_logger.WriteEntry("Fetching products from the database ...", LoggingLevel.Debug);
+
+                while (true)//to do
+                {
+                    //Getting products from DB
+                    var products = FetchProductsFromDb(dbContext, startIndex, _productsPerBatch);
+                    if (products == null) break;
+                    summary.ProductCount += products.Count;//products.Count = _productsPerBatch
+                    startIndex += products.Count;
+
+                    //_logger.WriteEntry($"{products.Count} products found...", LoggingLevel.Debug);
+
+                    foreach (var dbProd in products)
+                    {
+                        try
+                        {
+                            //what to do if product isn't available
+                            //what to do if price changes ==> up/down
+                            //attributes must be added the the amazon product
+                            //if analytics ==> price fluctuation/availability change rate/....
+
+                            ExecuteAndWait(() =>
+                            {
+                                var amzProd = _amazonClient.GetProductSummary(dbProd.Asin).Result;
+                                _logger.WriteEntry($"@Update#{_updatesCount} | @{dbProd.Asin} | Current Price : {dbProd.Price} |==> Amazon State : Price : {amzProd.Price} / Qty : {amzProd.Qty}", LoggingLevel.Debug);
+
+                                if (amzProd.IsUpdateRequired(dbProd))
+                                {
+                                    dbProd.Price = amzProd.Price;
+                                }
+
+                                dbContext.SaveChangesAsync();
```

Comment:
This should be inside the `if` block.  Outside the `if` block will result in attempts to save changes to the DB when there are no changes to save.